src/PayU/Alu/Response.php

```diff
@@ -472,36 +521,36 @@ private function computeInternalArray()
         $this->internalArray['RETURN_MESSAGE'] = $this->returnMessage;
         $this->internalArray['DATE'] = $this->date;
 
-        if (!is_null($this->amount)) {
+        if ($this->amount !== null) {
             $this->internalArray['AMOUNT'] = $this->amount;
```

Comment:
Would it be possible to remove usages of internalArray? So when computeInternalArray is being called, we can create, populate and return a new array.

I think the current implementation (meaning the old one) even has a bug.
Calling getResponseParams will initialize internalArray for the first time. But if a call a setter on the Response object and after that call getResponseParams  again, it will return the old state and not the new one

Reply:
I would not remove the $internalArray because it's used in a class from old namespace (Alu/Request) and it may break compatibility. Also, the only call of **getResponseParams** is made when the hash is validated (also a logic that cannot be changed because it's on Alu/HashService), so the scenario provided by you won't happen.